### ISSUE TITLE
OptimizerIrace instance handling

### ISSUE THREAD

**Issue by mb706** [mlr-org/bbotk#187]
Irace instance handling probably needs a bit more thought. Currently, the line

https://github.com/mlr-org/bbotk/blob/57b3d9ad23f591b00429a6494b5cfe7dde6b8e3c/R/OptimizerIrace.R#L182

adds the `instances` constant to an objective. This means
* the objective is changed by reference, probably not what one expects and probably not a good idea.
* if the objective already has the `instances` constant, then the irace call fails ("`Cannot add param with name 'instances': duplicated name`")
* in particular, if irace is called twice with the same objective, the second call fails since the first call already added the `instances` constant.

Maybe one way to solve this would be to assume that the `instances` constant must already be present to use irace? But idk if this breaks anything with tuning. It would at least solve the weird incongruency that, to use irace, one now needs to define a `ObjectiveRFunDt` with a `fun` that has an `instances` argument, but *without* specifying the `instances` constant.

Another solution would be to clone the objective and use that one inside OptimizerIrace. That could at least make repeated calls of `irace` work.

**Comment by be-marc** [comment]
Yes we could make it compulsory to create the objective with the `instances` constant:

```r
ObjectiveRFunDt$new(fun, domain, codomain, constants = ps(instances = p_uty())
```

I see no problem with tuning.

**Comment by MLopez-Ibanez** [comment]
Can we add something to irace to simplify this?

**Comment by mb706** [comment]
I think this is mostly an issue with the way we interface with irace. bbotk wasn't written with instances in mind and most of our optimizers don't have the concept, so we are still trying to find our way how to best represent instances in bbotk. The way irace does it looks fine to me. Thanks for asking, though! :+1:

**Comment by be-marc** [comment]
Closed by #200.
